utils/utils-lib/src/disk.rs

```diff
@@ -0,0 +1,35 @@
+use url::Url;
+
+/// Normalize the disks if they have a schema, we dont want to change anything
+/// or do any error checking -- the loop will converge to the error state eventually
+pub fn normalize_disk(disk: &str) -> String {
+    let disk = if disk.starts_with("pcie://") {
+        canonical_pcie_address(disk).unwrap_or_else(|| disk.to_string())
```

Comment:
if it's pcie can't we just return the `canonical_pcie_address` if we change that to return just the "normalized" content?

Reply:
I think it's better to keep the functionalities separate. Like `canonical_pcie_address` does only canonical conversion and `normalize_disk` does the normalize.